nucleus/admin/util/src/main/java/com/sun/enterprise/admin/remote/reader/ProgressStatusDTOJsonProprietaryReader.java

```diff
@@ -55,6 +58,7 @@ public ProgressStatusDTO readFrom(final InputStream is, final String contentType
             return readProgressStatus(jp);
         } finally {
             jp.close();
+            is.close();
```

Comment:
wondering about the same point:
https://github.com/eclipse-ee4j/glassfish/pull/24616#discussion_r1496778273

Reply:
Here it is because I tried to push the close method of the input stream closer to where it should be closed. Usually these input streams are byte array input streams or input streams from HttpUrlConnection, which should be closed too - in try-with, but that is impossible in current impl.

I wrote the javadoc for the readFrom method: 
`Reads and closes the stream, or can transfer both respnsibilities to a returned object.`

Originally some branches did not close the stream at all. Some branches even don't close the connection, seems it is left for garbage collector, but I am not sure with that. The old code is quite overcomplicated, hard to find all ends. So I am sure this is not the final version, rather a "milestone" of the refactoring. I would like to continue in smaller steps.